process_report/invoices/discount_invoice.py

```diff
@@ -0,0 +1,79 @@
+from dataclasses import dataclass
+
+import pandas
+
+import process_report.invoices.invoice as invoice
+
+
+@dataclass
+class DiscountInvoice(invoice.Invoice):
+    """
+    Invoice class containing functions useful for applying discounts
+    on dataframes
+    """
+
+    @staticmethod
+    def apply_flat_discount(
+        invoice: pandas.DataFrame,
+        pi_projects: pandas.DataFrame,
+        discount_amount: int,
+        discount_field: str,
+        balance_field: str,
+        code_field: str = None,
+        discount_code: str = None,
+    ):
+        """
+        Takes in an invoice and a list of PI projects that are a subset of it,
+        and applies a flat discount to those PI projects. Note that this function
+        will change the provided `invoice` Dataframe directly. Therefore, it does
+        not return the changed invoice.
+
+        This function assumes that the balance field shows the remaining cost of the project.
+        Basically, the provided invoice must show that balance = cost.
+
+        If the optional parameters `code_field` and `discount_code` are passed in,
+        `discount_code` will be comma-APPENDED to the `code_field` of projects where
+        the discount is applied
```

Comment:
@joachimweyl wouldn't it be a problem to have command separated field in what will be ultimately be a comma separated report?